airbyte-ci/connectors/metadata_service/orchestrator/orchestrator/assets/registry_report.py

```diff
@@ -22,148 +105,185 @@ def augment_and_normalize_connector_dataframes(
-    """
-    Generate a markdown report of the connector registry locations.
-    """
+    df.rename(columns=title_mapping, inplace=True)
 
-    columns_to_show = [
-        "dockerRepository",
-        "dockerImageTag",
-        "is_oss",
-        "is_cloud",
-        "is_source_controlled",
-        "is_spec_cached",
-        "is_metadata_valid",
-    ]
+    html_formatters = {column_info["title"]: column_info["formatter"] for column_info in column_mapping if "formatter" in column_info}
 
-    # convert true and false to checkmarks and x's
-    all_sources_dataframe.replace({True: "✅", False: "❌"}, inplace=True)
-    all_destinations_dataframe.replace({True: "✅", False: "❌"}, inplace=True)
+    columns = [column_info["title"] for column_info in column_mapping]
 
-    markdown = render_connector_registry_locations_markdown(
-        destinations_markdown=all_destinations_dataframe[columns_to_show].to_markdown(),
-        sources_markdown=all_sources_dataframe[columns_to_show].to_markdown(),
+    return df.to_html(
+        columns=columns, justify="left", index=False, formatters=html_formatters, escape=False, classes="styled-table", na_rep="❌", render_links=True
     )
 
-    registry_report_directory_manager = context.resources.registry_report_directory_manager
-    file_handle = registry_report_directory_manager.write_data(markdown.encode(), ext="md", key="connector_registry_locations")
 
-    metadata = {
-        "preview": MetadataValue.md(markdown),
-        "gcs_path": MetadataValue.url(file_handle.gcs_path),
-    }
-    return Output(metadata=metadata, value=file_handle)
+# ASSETS
+
+# TODO (ben): Update these assets to reference the new registry once deployed
 
 
 @asset(group_name=GROUP_NAME)
-def all_sources_dataframe(
-    cloud_sources_dataframe, oss_sources_dataframe, github_connector_folders, valid_metadata_report_dataframe, cached_specs
-) -> pd.DataFrame:
+def all_sources_dataframe(legacy_cloud_sources_dataframe, legacy_oss_sources_dataframe, github_connector_folders) -> pd.DataFrame:
     """
     Merge the cloud and oss sources registries into a single dataframe.
     """
 
     return augment_and_normalize_connector_dataframes(
-        cloud_df=cloud_sources_dataframe,
-        oss_df=oss_sources_dataframe,
+        cloud_df=legacy_cloud_sources_dataframe,
+        oss_df=legacy_oss_sources_dataframe,
         primaryKey="sourceDefinitionId",
         connector_type="source",
-        valid_metadata_report_dataframe=valid_metadata_report_dataframe,
         github_connector_folders=github_connector_folders,
-        cached_specs=cached_specs,
     )
 
 
 @asset(group_name=GROUP_NAME)
 def all_destinations_dataframe(
-    cloud_destinations_dataframe, oss_destinations_dataframe, github_connector_folders, valid_metadata_report_dataframe, cached_specs
+    legacy_cloud_destinations_dataframe, legacy_oss_destinations_dataframe, github_connector_folders
```

Comment:
As we might want to switch this back to the not legacy version soon, what do you think of:
- exposing a like `USE_LEGACY_REGISTRY` in this module,
- make `cloud_destinations_dataframe` and `oss_destinations_dataframe` assets  return the legacy DF if `USE_LEGACY_REGISTRY is True`.

#feature flags 😆

Reply:
Very good idea!

Reply:
Looking into it, we'd still have to update the sensor. Could still be flagged but I think this juice might not be worth the squeeze as its an informational report and not critical to rollback if something goes wrong